crates/ruff_formatter/src/builders.rs

```diff
@@ -2569,25 +2569,20 @@ impl<Context> Format<Context> for BestFitting<'_, Context> {
     fn fmt(&self, f: &mut Formatter<Context>) -> FormatResult<()> {
         let variants = self.variants.items();
 
-        let mut formatted_variants = Vec::with_capacity(variants.len());
+        let mut buffer = VecBuffer::with_capacity(variants.len() * 8, f.state_mut());
```

Comment:
It would be nice to remove the need for this buffer too but this complicates things a bit because finding the boundaries of a variant suddenly need to account for nested best fitting elements:

```
StartBestFittingEntry (Outer)
....
StartBestFittingEntry (Inner)
...
EndBestFittingEntry
...
EndBestFittingEntry
```

The nesting is probably rare because best fitting is almost always used together with `Interned`, meaning that the nested best fitting most likely ends up being inside of the interned vec, but it remains possible.